packages/utils/src/lib/filter-by-slug.ts

```diff
@@ -0,0 +1,39 @@
+import { Audit, Group } from '@code-pushup/models';
+import { toArray } from './transform';
+
+export function filterByAuditSlug(
+  groups: Group[],
+  auditSlugs: string | string[],
+): Group[] {
+  const slugs = toArray(auditSlugs);
+  if (slugs.length === 0) {
+    return groups;
+  }
+  return (
+    groups
+      .map(group => ({
+        ...group,
+        refs: filterSlug(group.refs, slugs),
+      }))
+      // filter out groups that have no audits includes from onlyAudits (avoid empty groups)
+      .filter(group => group.refs.length)
+  );
+}
+
+export function filterBySlug(
+  list: Audit[],
+  auditSlugs: string[] | string,
+): Audit[] {
+  const slugs = toArray(auditSlugs);
+  if (slugs.length === 0) {
+    return list;
+  }
+  return filterSlug(list, slugs);
+}
+
+export function filterSlug<T extends { slug: string }>(
+  refs: T[],
+  slugs: string | string[],
+): T[] {
+  return refs.filter(({ slug }) => slugs.includes(slug));
+}
```

Comment:
:lady_beetle:  Careful, `slugs` is a single string and not an array, then `includes` will match substrings, which is not what you want.

```suggestion
export function filterSlug<T extends { slug: string }>(
  refs: T[],
  slugOrSlugs: string | string[],
): T[] {
  const slugs = toArray(slugOrSlugs);
  return refs.filter(({ slug }) => slugs.includes(slug));
}
```

or

```suggestion
export function filterSlug<T extends { slug: string }>(
  refs: T[],
  slugOrSlugs: string | string[],
): T[] {
  return refs.filter(({ slug }) =>
    Array.isArray(slugOrSlugs) ? slugOrSlugs.includes(slug) : slug === slugOrSlugs,
  );
}
```